### DISCUSSION TITLE
Example for Passwordless and Usernameless?

### DISCUSSION THREAD

**Discussion by Oba-One** [MasterKale/SimpleWebAuthn#373] [Troubleshooting]
Hi! Looking to implement the server but not for 2FA is there an example I can follow? On the site it says 
`Guides for "Passwordless" and "Usernameless" support are coming soon.`

**Comment by MasterKale** [comment]
Hello @Oba-One, here's my guide to implementing passkeys support, in the docs website, that should be a good starting point:

https://simplewebauthn.dev/docs/advanced/passkeys

Requiring User Verification will be what achieves passwordless use of WebAuthn. And requiring discoverable credentials will help you get to usernameless via conditional UI:

https://simplewebauthn.dev/docs/packages/browser#browser-autofill-aka-conditional-ui

Please let me know if anything in either of these sections needs clarification.

And can you remind me where you saw the mention of passwordless and usernameless guides "coming soon?" I should probably update that to point to these guides 😅

**Reply by Oba-One** [reply]
Thanks for the help, here's where I found it. https://simplewebauthn.dev/docs/packages/server/#identifying-your-rp

**Reply by Oba-One** [reply]
@MasterKale How should I be storing the credential id and public key in the database? Should it be converted from a Uint8 array to buffer then string or stored as an array? I'm using a postgres database for context and want to use the id and the primary key.

**Reply by MasterKale** [reply]
I defined an Authenticator type here with some hints for how to map the various JS values to typical RDBMS column types:

https://simplewebauthn.dev/docs/packages/server#installing

Pay attention to the comments that start with "SQL", those should help you figure out types that work best for you.

**Reply by Oba-One** [reply]
Yep I saw that, my confusion comes from the docs saying Buffer and the type in code being an Uint8Array. For now I'll convert to a buffer and then to string.
